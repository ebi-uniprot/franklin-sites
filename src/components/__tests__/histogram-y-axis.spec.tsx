import { scaleLinear } from 'd3';
import { render, screen } from '@testing-library/react';
import YAxis from '../histogram-y-axis';

describe('XAxis component', () => {
  it('should render with correct tickmarks', () => {
    const max = 10;
    const scale = scaleLinear().domain([0, max]).range([100, 0]);
    const { asFragment } = render(<YAxis scale={scale} height={100} />);
    expect(asFragment()).toMatchSnapshot();
    const labels = screen
      .getAllByText(/\d+/)
      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
      // @ts-ignore
      .map(({ __data__: label }) => label);
    expect(labels).toEqual([0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10]);
  });
});
