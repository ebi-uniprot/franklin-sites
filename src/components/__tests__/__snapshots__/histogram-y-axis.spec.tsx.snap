// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`XAxis component should render with correct tickmarks 1`] = `
<DocumentFragment>
  <svg
    class="y-axis"
    height="100"
    width="80"
  >
    <g
      fill="none"
      font-family="sans-serif"
      font-size="10"
      text-anchor="end"
      transform="translate(50, 0)"
    >
      <path
        class="domain"
        d="M0.5,100.5V0.5"
        stroke="currentColor"
      />
      <g
        class="tick"
        opacity="1"
        transform="translate(0,100.5)"
      >
        <line
          stroke="currentColor"
          x2="0"
        />
        <text
          dy="0.32em"
          fill="currentColor"
          x="-6"
        >
          0
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,90.5)"
      >
        <line
          stroke="currentColor"
          x2="0"
        />
        <text
          dy="0.32em"
          fill="currentColor"
          x="-6"
        >
          1
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,80.5)"
      >
        <line
          stroke="currentColor"
          x2="0"
        />
        <text
          dy="0.32em"
          fill="currentColor"
          x="-6"
        >
          2
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,70.5)"
      >
        <line
          stroke="currentColor"
          x2="0"
        />
        <text
          dy="0.32em"
          fill="currentColor"
          x="-6"
        >
          3
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,60.5)"
      >
        <line
          stroke="currentColor"
          x2="0"
        />
        <text
          dy="0.32em"
          fill="currentColor"
          x="-6"
        >
          4
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,50.5)"
      >
        <line
          stroke="currentColor"
          x2="0"
        />
        <text
          dy="0.32em"
          fill="currentColor"
          x="-6"
        >
          5
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,40.5)"
      >
        <line
          stroke="currentColor"
          x2="0"
        />
        <text
          dy="0.32em"
          fill="currentColor"
          x="-6"
        >
          6
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,30.500000000000004)"
      >
        <line
          stroke="currentColor"
          x2="0"
        />
        <text
          dy="0.32em"
          fill="currentColor"
          x="-6"
        >
          7
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,20.499999999999996)"
      >
        <line
          stroke="currentColor"
          x2="0"
        />
        <text
          dy="0.32em"
          fill="currentColor"
          x="-6"
        >
          8
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,10.499999999999998)"
      >
        <line
          stroke="currentColor"
          x2="0"
        />
        <text
          dy="0.32em"
          fill="currentColor"
          x="-6"
        >
          9
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,0.5)"
      >
        <line
          stroke="currentColor"
          x2="0"
        />
        <text
          dy="0.32em"
          fill="currentColor"
          x="-6"
        >
          10
        </text>
      </g>
    </g>
  </svg>
</DocumentFragment>
`;
